Sources/MapboxMaps/Foundation/Camera/GestureDecelerationCameraAnimator.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+
+internal final class GestureDecelerationCameraAnimator: NSObject, CameraAnimatorInterface {
+
+    private var location: CGPoint
+    private var velocity: CGPoint
+    private let decelerationRate: CGFloat
+    private let locationChangeHandler: (CGPoint) -> Void
+    private var previousDate: Date?
+    private let dateProvider: DateProvider
+    internal var completion: (() -> Void)?
+
+    internal init(location: CGPoint,
+                  velocity: CGPoint,
+                  decelerationRate: CGFloat,
+                  locationChangeHandler: @escaping (CGPoint) -> Void,
+                  dateProvider: DateProvider) {
+        self.location = location
+        self.velocity = velocity
+        self.decelerationRate = decelerationRate
+        self.locationChangeHandler = locationChangeHandler
+        self.dateProvider = dateProvider
+    }
+
+    internal private(set) var state: UIViewAnimatingState = .inactive
+
+    internal func cancel() {
+        stopAnimation()
+    }
+
+    internal func startAnimation() {
+        previousDate = dateProvider.now
+        state = .active
+    }
+
+    internal func stopAnimation() {
+        state = .inactive
+        completion?()
+        completion = nil
+    }
+
+    internal func update() {
+        guard state == .active, let previousDate = previousDate else {
+            return
+        }
+
+        let currentDate = dateProvider.now
+        self.previousDate = currentDate
```

Comment:
Might want to investigate moving to `CACurrentMediaTime` here too as part of https://github.com/mapbox/mapbox-maps-ios/issues/293

Reply:
Yep. Okay if we add this to that ticket and investigate it when we get to that?

Reply:
Yep.